lua/wikis/lab/MatchGroup/Input/Custom.lua

```diff
@@ -0,0 +1,98 @@
+---
+-- @Liquipedia
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Operator = require('Module:Operator')
+local Variables = require('Module:Variables')
+
+local MatchGroupInputUtil = Lua.import('Module:MatchGroup/Input/Util')
+
+local DEFAULT_BESTOF = 3
+
+local CustomMatchGroupInput = {}
+local MatchFunctions = {
+	DEFAULT_MODE = 'team',
+}
+local MapFunctions = {}
+local FfaMatchFunctions = {
+	DEFAULT_MODE = 'team',
+}
+local FfaMapFunctions = {}
+
+---@param match table
+---@param options table?
+---@return table
+function CustomMatchGroupInput.processMatch(match, options)
+	return MatchGroupInputUtil.standardProcessMatch(match, MatchFunctions, FfaMatchFunctions)
+end
+
+-- "Normal" match
+---@param match table
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	return MatchGroupInputUtil.standardProcessMaps(match, opponents, MapFunctions)
+end
+
+---@param games table[]
+---@return table[]
+function MatchFunctions.removeUnsetMaps(games)
+	return Array.filter(games, function(map)
+		return map.map ~= nil
+	end)
+end
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
+end
+
+---@param bestofInput string|integer?
+---@return integer?
+function MatchFunctions.getBestOf(bestofInput)
+	local bestof = tonumber(Logic.emptyOr(bestofInput, Variables.varDefault('bestof')))
+	Variables.varDefine('bestof', bestof)
+	return bestof or DEFAULT_BESTOF
+end
```

Comment:
sure this is the behaviour we want for this wiki?
i think most wikis use the number of maps = bestof approach